middle_end/flambda2/from_lambda/lambda_to_flambda.ml

```diff
@@ -1428,7 +1480,7 @@ and cps_function env ~fid ~(recursive : Recursive.t) ?precomputed_free_idents
     ~params_arity ~removed_params ~return ~calling_convention
     ~return_continuation:body_cont ~exn_continuation ~my_region ~my_ghost_region
     ~body ~attr ~loc ~free_idents_of_body recursive ~closure_alloc_mode:mode
-    ~first_complex_local_param ~contains_no_escaping_local_allocs:region
```

Comment:
I think we want to allow functions with `contains_no_escaping_local_allocs = true` but `ret_mode = Alloc_local`, when the result is not actually allocated on the stack.

Reply:
In flambda2, maybe, but we can't rely on `region` for that (it only means that a region was inserted, but with `Lexclave` that doesn't mean that no allocations escape).
So we cannot set the `contains_no_escaping_local_allocations` flag to `true` when the return mode is `Alloc_local` unless we perform a dedicated analysis.

Reply:
Please note that #2985 which we will merge very soon, removes `contains_no_escaping_local_allocations` - would that simplify or complicate this discussion?

Reply:
In a way, this PR re-introduces it under another name: the `my_region` parameter is now optional, if it is absent it means that the function doesn't contain any escaping local allocations. So we still conservatively put a `my_region` parameter whenever the return mode is `Local`, and could remove it with an analysis if it turns out to be not necessary.